server/src/services/users/users.go

```diff
@@ -74,5 +78,28 @@ func (s *UserService) Update(_ context.Context, body dto.UserUpdateRequest) (*dt
 		Name:   body.Name,
 		Avatar: body.Avatar,
 	})
+	s.UpdatedUser(user)
+
 	return new(dto.User).From(user), err
 }
+
+func (s *UserService) UpdatedUser(user database.User) {
+	connectedBoards, err := s.database.GetSingleUserConnectedBoards(user.ID)
+	if err != nil {
+		return
+	}
+	for _, session := range connectedBoards {
+		userSession, err := s.database.GetBoardSession(session.Board, session.User)
+		if err != nil {
+			logger.Get().Errorw("unable to get board session", "board", userSession.Board, "user", userSession.User.ID(), "err", err)
+		}
+		err = s.realtime.BroadcastToBoard(session.Board, realtime.BoardEvent{
+			Type: realtime.BoardEventParticipantUpdated,
+			Data: new(dto.BoardSession).From(session),
+		})
+
+		if err != nil {
+			logger.Get().Errorw("unable to broadcast updated user", "err", err)
+		}
+	}
```

Comment:
If I understand correctly, every time a user's role is updated, a ParticipantUpdated event is sent to all boards the user is connected to?

That is, if a user is currently connected to multiple boards and their role is updated on one board, the event will still be sent to all boards.

Reply:
ParticipantUpdated is not only called, when a user is promoted or demoted but also when he changes his appearance (avatar), name etc.

This is why we need to get all currently connected boards of a user and update the changes on all of them.

When we loop over the connectedBoards, each session is retrieved from the database one by one. This is why a name change would be published to all currently connected boards, but a ban would only take effect on a specific board.

Reply:
It returns the same information as before which therefore should be okay. 😄